timemachine/fe/single_topology_v3.py

```diff
@@ -483,6 +484,221 @@ def find_dummy_groups_and_anchors(mol_a, mol_b, core_a, core_b):
     return dummy_groups_b, all_jks
 
 
+def handle_ring_opening_closing(
+    f: Callable[[float, float, float], float],
+    src_k: float,
+    dst_k: float,
+    lamb: float,
+    lambda_min: float,
+    lambda_max: float,
+):
+    """
+    In the typical case (src_k != 0 and dst_k != 0), use the specified interpolation function, f.
+
+    In the case where src_k = 0 or dst_k = 0 (e.g. ring closing and ring opening, respectively), restrict interpolation
+    to the interval [lambda_min, lambda_max], and pin to the end state values outside of this range.
+
+    Parameters
+    ----------
+    f : callable, (src_k: float, dst_k: float, lam: float) -> float
+        interpolation function; should satisfy f(0) = src_k, f(1) = dst_k
+
+    src_k, dst_k : float, k >= 0
+        force constants at lambda=0 and lambda=1, respectively
+
+    lambda_min, lambda_max : float, in 0 < lambda_min < lambda_max < 1
+        interpolate in range [lambda_min, lambda_max] (pin to end states otherwise). Note that if dst_k=0, the
+        convention is flipped so that 1 - lambda_min corresponds to f(0) and 1 - lambda_max corresponds to f(1).
```

Comment:
Missing return description

Reply:
Added in [df8632](https://github.com/proteneer/timemachine/pull/820/commits/cdf8632e032a7057ad4ba20f6c74cf7c50224c7f)